src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java

```diff
@@ -1,29 +1,47 @@
 package ch.wisv.areafiftylan.service;
 
+import ch.wisv.areafiftylan.dto.TeamDTO;
 import ch.wisv.areafiftylan.model.Team;
+import ch.wisv.areafiftylan.model.User;
 import ch.wisv.areafiftylan.service.repository.TeamRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.Collection;
+import java.util.Optional;
 
 @Service
 public class TeamServiceImpl implements TeamService {
     private final TeamRepository teamRepository;
+    private final UserService userService;
 
     @Autowired
-    public TeamServiceImpl(TeamRepository teamRepository) {
+    public TeamServiceImpl(TeamRepository teamRepository, UserService userService) {
         this.teamRepository = teamRepository;
+        this.userService = userService;
     }
 
     @Override
-    public Team getTeamById(Long id) {
-        return teamRepository.findOne(id);
+    public Team create(TeamDTO teamDTO) {
+        User captain = userService.getUserById(teamDTO.getCaptainID()).get();
+        Team team = new Team(teamDTO.getTeamName(), captain);
+
+        return teamRepository.saveAndFlush(team);
+    }
+
+    @Override
+    public Team save(Team team) {
+        return teamRepository.save(team);
     }
 
     @Override
-    public Team getTeamByTeamname(String teamname) {
-        return teamRepository.findByTeamName(teamname);
+    public Optional<Team> getTeamById(Long id) {
+        return Optional.ofNullable(teamRepository.findOne(id));
```

Comment:
Ik zou dingen niet handmatig in een Optional wrappen, maar dit vanuit de Repository doen mocht dat nodig zijn.